tests/test_download.py

```diff
@@ -118,18 +146,18 @@ def test_download_url_to_tempfileobj_and_lengths(self):
     # the server-reported length of the file does not match the
     # required_length.  'updater.py' *does* verify the hashes of downloaded
     # content.
-    download.safe_download(self.url, self.target_data_length - 4).close()
-    download.unsafe_download(self.url, self.target_data_length - 4).close()
+    download.safe_download(self.url, self.target_data_length - 4, self.fetcher).close()
+    download.unsafe_download(self.url, self.target_data_length - 4, self.fetcher).close()
 
     # We catch 'tuf.exceptions.DownloadLengthMismatchError' for safe_download()
     # because it will not download more bytes than requested (in this case, a
     # length greater than the size of the target file).
     self.assertRaises(tuf.exceptions.DownloadLengthMismatchError,
-        download.safe_download, self.url, self.target_data_length + 1)
+        download.safe_download, self.url, self.target_data_length + 1, self.fetcher)
```

Comment:
should there be a `.close()` call on the downloaded file here, like the calls to `safe_download()` above and below?

Reply:
Ah, now I see that we close the file when we catch an exception. Hmm, I wonder if that's expected (anticipated) behaviour?

Reply:
In the exception case there is no file to close at this point, I think this is correct (it just looks weird in the tests since we don't ever do anything with the file -- normally you'd use a `with download.safe_download() as f:` construct)